src/core/toast.ts

```diff
@@ -195,7 +195,13 @@ toast.dark = (content: ToastContent, options?: ToastOptions) =>
 /**
  * Remove toast programmaticaly
  */
-toast.dismiss = (id?: Id) => eventManager.emit(Event.Clear, id);
+toast.dismiss = (id?: Id) => {
+  if (containers.size > 0) {
+    eventManager.emit(Event.Clear, id);
+  } else {
+    queue = queue.filter(t => id !== undefined && t.options.toastId !== id);
```

Comment:
I would replace `id !== undefined` by `id != undefined`. Just to check for both undefined and null.
```suggestion
    queue = queue.filter(t => id != undefined && t.options.toastId !== id);
```

Reply:
Actually, even this code will do the work 🤔? WDYT?
```suggestion
    queue = queue.filter(t => t.options.toastId !== id);
```

Reply:
I was trying to preserve the behavior of `toast.dismiss`. I gathered that this function would remove all toasts if you did not provide an id. Therefore the whole queue will be emptied if `id` is undefined.

Reply:
It might be better/more consistent to use `isToastIdValid`, though?

Reply:
Indeed you're right.